### PR TITLE
 feature/Task-4  Implemented caching logic for Event resource

### PR DESCRIPTION
**Pull request by ValeryJK**

### CODE REVIEW

File: src/EventBookSystem.API/Controllers/EventsController.cs [hunk 1]
```diff
@@ -13,31 +14,54 @@ namespace EventBookSystem.API.Controllers
...
+            _cache = cache;
+        }
 
         [HttpGet]
+        [ResponseCache(Duration = 60, Location = ResponseCacheLocation.Client, NoStore = false)]
         public async Task<IActionResult> GetAllEvents()
         {
-            var events = await _eventService.GetAllEventsAsync();
+            var cacheKey = "AllEvents";
+            if (!_cache.TryGetValue(cacheKey, out IEnumerable<EventDto>? events))
```
Comment (D9bolic) [inline src/EventBookSystem.API/Controllers/EventsController.cs:30]: please use [GetOrCreateAsync]

(https://learn.microsoft.com/en-us/dotnet/api/microsoft.extensions.caching.memory.cacheextensions.getorcreateasync?view=net-8.0#microsoft-extensions-caching-memory-cacheextensions-getorcreateasync-1(microsoft-extensions-caching-memory-imemorycache-system-object-system-func((microsoft-extensions-caching-memory-icacheentry-system-threading-tasks-task((-0))))))

File: src/EventBookSystem.API/Controllers/EventsController.cs [hunk 2]
```diff
@@ -13,31 +14,54 @@ namespace EventBookSystem.API.Controllers
...
+            {
+                events = await _eventService.GetAllEventsAsync();
+                var cacheEntryOptions = new MemoryCacheEntryOptions()
+                    .SetSlidingExpiration(TimeSpan.FromMinutes(5));
```
Comment (D9bolic) [inline src/EventBookSystem.API/Controllers/EventsController.cs:34]: 1) magic number
2) Shall be configurable via appsetting json
3) Service layer looks like better solution for the caching logic

File: src/EventBookSystem.API/Controllers/EventsController.cs [hunk 3]
```diff
@@ -13,31 +14,54 @@ namespace EventBookSystem.API.Controllers
...
+
+                _cache.Set(cacheKey, events, cacheEntryOptions);
+            }
 
             return Ok(events);
         }
 
         [HttpGet("{eventId}")]
+        [ResponseCache(Duration = 60, Location = ResponseCacheLocation.Client, NoStore = false)]
```
Comment (D9bolic) [inline src/EventBookSystem.API/Controllers/EventsController.cs:43]: Duration as const

File: src/EventBookSystem.API/Controllers/EventsController.cs [hunk 4]
```diff
@@ -13,31 +14,54 @@ namespace EventBookSystem.API.Controllers
...
+            _eventService = eventService;
+            _cache = cache;
+        }
 
         [HttpGet]
+        [ResponseCache(Duration = 60, Location = ResponseCacheLocation.Client, NoStore = false)]
         public async Task<IActionResult> GetAllEvents()
         {
-            var events = await _eventService.GetAllEventsAsync();
+            var cacheKey = "AllEvents";
```
Comment (D9bolic) [inline src/EventBookSystem.API/Controllers/EventsController.cs:29]: magic number - to const

File: src/EventBookSystem.API/Controllers/EventsController.cs [hunk 5]
```diff
@@ -55,6 +79,8 @@ public async Task<IActionResult> CreateEventAsync([FromBody] EventForCreationDto
 
             var createdEvent = await _eventService.CreateEventAsync(eventDto);
 
+            await InvalidateEventCache();
```
Comment (D9bolic) [inline src/EventBookSystem.API/Controllers/EventsController.cs:82]: part of service scenario

File: src/EventBookSystem.API/Controllers/EventsController.cs [hunk 6]
```diff
@@ -80,5 +109,11 @@ public async Task<IActionResult> DeleteEventAsync(Guid eventId)
 
             return NoContent();
         }
+
+        private async Task InvalidateEventCache()
+        {
+            _cache.Remove("AllEvents");
```
Comment (D9bolic) [inline src/EventBookSystem.API/Controllers/EventsController.cs:115]: whay is it task?

**Review by D9bolic — CHANGES_REQUESTED** [review]

File: src/EventBookSystem.API/Controllers/EventsController.cs (hunk 1 above)
Comment (ValeryJK) [reply src/EventBookSystem.API/Controllers/EventsController.cs]: done

File: src/EventBookSystem.API/Controllers/EventsController.cs (hunk 2 above)
Comment (ValeryJK) [reply src/EventBookSystem.API/Controllers/EventsController.cs]: done

File: src/EventBookSystem.API/Controllers/EventsController.cs (hunk 3 above)
Comment (ValeryJK) [reply src/EventBookSystem.API/Controllers/EventsController.cs]: done

File: src/EventBookSystem.API/Controllers/EventsController.cs (hunk 4 above)
Comment (ValeryJK) [reply src/EventBookSystem.API/Controllers/EventsController.cs]: done

File: src/EventBookSystem.API/Controllers/EventsController.cs (hunk 5 above)
Comment (ValeryJK) [reply src/EventBookSystem.API/Controllers/EventsController.cs]: done

File: src/EventBookSystem.API/Controllers/EventsController.cs (hunk 6 above)
Comment (ValeryJK) [reply src/EventBookSystem.API/Controllers/EventsController.cs]: done

**Pull request merged**